### PR TITLE
fixes for web and added portal

### PR DESCRIPTION
**Pull request by inayat1**
Change log - 
1). Moved it to es6
2). In calculateModalPosition function, removed the condition of checking for the container and content height. If both height are same then we were giving position equal to container height i.e. modal will be transformed with container height. But ideally it should have transformed with 0px(show in full page). So removed it as it was breaking for the bottomsheet where content height was equal to containerheight
3). Added portals.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -226,33 +189,28 @@ var ModalBox = createReactClass({
...
 
     /*
      * Calculate when should be placed the modal
      */
-    calculateModalPosition: function (containerHeight, containerWidth) {
+    calculateModalPosition = (containerHeight, containerWidth) => {
         var position = 0;
-
         if (this.props.position == 'bottom') {
-            if (containerHeight != this.state.height) {
```
Comment (surya-kanoria) [inline index.js:238]: Please check if this change does not break anything.

File: index.js [hunk 2]
```diff
@@ -378,21 +330,23 @@ var ModalBox = createReactClass({
...
-            <View style={[styles.transparent, styles.absolute]} pointerEvents={'box-none'} onLayout={this.onContainerLayout}>
-                {backdrop}
-                <Animated.View
-                    onLayout={this.onViewLayout}
-                    style={[styles.wrapper, size, this.props.style, { transform: [{ translateY: this.state.position }, { translateX: offsetX }] }, Platform.OS === 'web' ? { willChange: 'transform' } : {}]}
-                    {...this.state.pan.panHandlers}>
-                    {this.props.children}
-                </Animated.View>
-            </View>
+            <Portal>
```
Comment (surya-kanoria) [inline index.js]: How will Portal work for android and iOS?

File: index.js [hunk 3]
```diff
@@ -35,46 +33,11 @@ var styles = StyleSheet.create({
...
-    //   backdropColor: React.PropTypes.string,
-    //   backdropContent: React.PropTypes.element,
-    //   animationDuration: React.PropTypes.number,
-    //
-    //   onClosed: React.PropTypes.func,
-    //   onOpened: React.PropTypes.func,
-    //   onClosingState: React.PropTypes.func,
-    // },
-
-    getDefaultProps: function () {
```
Comment (surya-kanoria) [inline index.js]: Please see what happens if default props are removed.

File: index.js [hunk 4]
```diff
@@ -26,7 +24,7 @@ var styles = StyleSheet.create({
     },
 
     absolute: {
-        position: "absolute",
+        position: "fixed",
```
Comment (surya-kanoria) [inline index.js]: Sometimes modal box are used inside a React component. Making position to "fixed", removes the flexibility. This seems unnecessary.

Comment (inayat1) [reply index.js]: changed to absolute again.

File: index.js (hunk 2 above)
Comment (inayat1) [reply index.js]: This is only for web, not used by android or ios

File: index.js (hunk 1 above)
Comment (inayat1) [reply index.js:238]: Muskein was facing some issue in grocery thats y he added the check. But this breaks for full height bottom sheet. So for now sending a prop resizeToFullscreen to add a check.

File: index.js (hunk 2 above)
Comment (inayat1) [reply index.js]: Added a check for portals using prop showFullscreen

**Pull request merged**